hack/build-clusteradm-image.sh

```diff
@@ -73,11 +74,8 @@ echo "Using clusteradm v${clusteradm_version}."
 
 git clone -b "v$clusteradm_version" --depth 1 https://github.com/open-cluster-management-io/clusteradm.git "$clusteradm_folder"
 
-cd "$clusteradm_folder"
-
-export KO_DOCKER_REPO=$registry
-
-ko build -B ./cmd/clusteradm -t $clusteradm_version --sbom=none --platform $platform
-
-cd ~
-rm -rf $clusteradm_folder
+( # Ensure that the temprary folder is deketed even on error
```

Comment:
(A) typo in "deketed".

(B) the `trap 'rm ...' EXIT` ensures that the `rm` command is executed no matter what.

(C) The parentheses here add nothing. Further, they are harmful because they cause a subshell to be used, so all the local shell variables are lost. It would be better to use curly braces rather than parentheses, if anything.

(D) I am not a fan of deleting the evidence when something goes wrong. In my opinion, it would be better to `{ ... commands to build ...} && rm -rf "$clusteradm_folder"` so that the evidence goes away only on success.

Reply:
@MikeSpreitzer 

(B) if the script fails on the `ko` command, because if the `set -e` it will terminate and exit immediately, the `trap` will indeed trigger but since the script has `cd` to the temp folder that would like to remove, it will NOT remove it. I did observe this behavior and took actions

(C) indeed the parenthesis add the extra shell execution so that if it fails, the "main" script will NOT be holding the temp folder, it will still exit immediately due to `set -e` but the `trap` will work in this case... also tested it

(D) the temp folder does not contain anything useful as it is an exact copy of somebody else repo. 99% of ko fail are related to the user having forgotten to login into quay.io before running the script. I'm not a fan to keep garbage in `.`. I would be ok to let it be in `/tmp` as it would be eventually deleted by the system. A second successful run of the script after a previous failure will leave the previous temp folder in place

Reply:
@MikeSpreitzer I just poshed a new version with a different solution to the problem that does nor require the temporary shell

Reply:
Oh, right, the parentheses _were_ doing something.

Since there is so little distinct information in the build directory, I am OK with it being deleted even on failure.

The build directory does contain one bit of information not in Git: _which_ version was attempted to be built. Maybe if the script echoed that, there would be no doubt.